cmake/target/version/generate_version_info.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+"""
+# ===============================================================================
+# NAME: generate_version_info.py
+#
+# DESCRIPTION:  Creates a version.hpp and version.json file with version informations
+#
+# USAGE: ./generate_version_info.py /path/to/outdir/
+#
+# AUTHOR: sfregoso
+# EMAIL:  sfregoso@jpl.nasa.gov
+# DATE CREATED  : Oct. 15, 2021
+#
+# Copyright 2021, California Institute of Technology.
+# ALL RIGHTS RESERVED. U.S. Government Sponsorship acknowledged.
+# ===============================================================================
+"""
+import argparse
+import os
+import sys
+import json
+from pathlib import Path
+
+from fprime_ac.utils.version import (
+    FALLBACK_VERSION,
+    get_fprime_version,
+    get_project_version,
+    get_library_versions,
+)
+
+
+def create_version_file_hpp(output_dir, framework_version, project_version):
+    """
+    Create the version file using the provided name and path.
+    """
+    version_hpp = Path(output_dir) / "version.hpp.tmp"
+    # Open file for writing
+    with open(version_hpp, "w") as fid:
+        fid.write("/*\n")
+        fid.write(
+            f"    This file has been autogenerated using [{os.path.basename(__file__)}].\n"
+        )
+        fid.write("    This file may be overwritten.\n")
+        fid.write("*/\n")
+        fid.write("#ifndef _VERSION_HPP_\n")
+        fid.write("#define _VERSION_HPP_\n")
+        fid.write("\n")
+        fid.write(f'static const char* FRAMEWORK_VERSION = "{framework_version}";\n')
+        fid.write(f'static const char* PROJECT_VERSION = "{project_version}";\n')
+        if len(get_library_versions()) > 0:
```

Comment:
If there are no libraries, perhaps add an else with dummy string for lib name and version that says no lib versions available?

Reply:
If you are ok parsing that in your Version component, that works for me!

Reply:
I think I will take the strings as-is and put them on an EVR. It's not like we use the values to determine anything. It could be something like {"LIB_NA"}@{"Version_NA"}. Unless @LeStarch has other ideas?

Reply:
The current structure has them being single strings, separated with `@`
```
LIBRARY_VERSIONS[] = {
      "lib1@v1",
      "lib2@v2"
}
```
I pushed the following version for when there is no libraries
```
LIBRARY_VERSIONS[] = { "" }
```
but am happy to switch if we think there is a better approach.

Reply:
Scratch that.  Can we do:

LIBRARY_VERSIONS[] = { nullptr }

instead?  Make the check easier.  See below.

Reply:
Done